src/lib.rs

```diff
@@ -394,35 +399,94 @@ impl Check {
                                 Some(version),
                             )?;
 
-                            Ok(run_check_release(
-                                &mut config,
-                                crate_name,
-                                current_crate,
-                                baseline_crate,
-                                self.release_type,
-                            )?)
+                            Ok((
+                                crate_name.clone(),
+                                Some(run_check_release(
+                                    &mut config,
+                                    crate_name,
+                                    current_crate,
+                                    baseline_crate,
+                                    self.release_type,
+                                )?),
+                            ))
                         }
                     })
                     .collect()
             }
         };
-        let success = all_outcomes
-            .into_iter()
-            .fold_ok(true, std::ops::BitAnd::bitand)?;
+        let crate_reports: BTreeMap<String, CrateReport> = {
+            let mut reports = BTreeMap::new();
+            for outcome in all_outcomes {
+                let (name, outcome) = outcome?;
+                if let Some(outcome) = outcome {
+                    reports.insert(name, outcome);
+                }
+            }
+            reports
+        };
+
+        Ok(Report { crate_reports })
+    }
+}
+
+/// Report of semver check of one crate.
+#[non_exhaustive]
+#[derive(Debug)]
+pub struct CrateReport {
+    /// Bump between the current version and the baseline one.
+    detected_bump: ActualSemverUpdate,
+    /// Minimum bump required to respect semver.
+    /// For example, if the crate contains breaking changes, this is [`ReleaseType::Major`].
+    /// If no bump is required, this is [`Option::None`].
```

Comment:
Is this accurate?

Say we ran a semver-check over a crate where the detected bump is `ActualSemverUpdate::Major`. Would `required_bump` actually be `Some(ReleaseType::Major)`?

If no bump is required, the value will be `None`, but is that the only time? I think there are other situations where the value might be `None`.

Let's back this up with some tests? Consider making a new integration test file in the `tests/` directory to pin down the behavior here.